mlir_graphblas/src/lib/GraphBLAS/GraphBLASPasses.cpp

```diff
@@ -171,7 +171,9 @@ class LowerTransposeRewrite : public OpRewritePattern<graphblas::TransposeOp> {
 
     rewriter.create<memref::StoreOp>(loc, last_last, outputPtrs, ncol);
 
-    rewriter.replaceOp(op, output);
+    Type cscTensor = getCSCTensorType(context, {-1, -1}, valueType);
+    Value newOutput = rewriter.create<tensor::CastOp>(loc, cscTensor, output);
```

Comment:
This assumes the input is CSR and output is CSC. We should key off the result type because the call might be CSC->CSR.

Reply:
I've fixed this, though it required doing a little tensor casting around our external C functions (like dup_tensor, empty_like, etc) because we need a consistent signature for those functions.  Since they ultimately receive the tensor as an opaque pointer, I temporarily cast the array to CSR of a specific type to call them.  For the two that return a tensor, I cast the tensor back to match the type of the input tensor.